### ISSUE TITLE
RuntimeError: [enforce fail at operator.cc:52] blob != nullptr. op Scale: Encountered a non-existing input blob: 1

### ISSUE THREAD

**Issue by christoph-conrads** [onnx/onnx-caffe2#48]
I have a simple machine learning model for MNIST consisting of a single 748x10 fully connected layer. I can export the model from [PyTorch commit a0ac72e](https://github.com/pytorch/pytorch/tree/a0ac72e84e2f14c42d1bfdd695c411c0f101487c), I can convert the ONNX data to Caffe2 with 
```sh
convert-onnx-to-caffe2 --output MNIST.simple.caffe2 --init-net-output MNIST.simple.init.caffe2 -- MNIST.simple.proto
```
using [onnx-caffe2 8a40069a](https://github.com/onnx/onnx-caffe2/tree/8a40069a72a05c46fa9e66bb63d843d95aa3f0f9), and I can run inference on the exported model using onnx-caffe2 but I cannot run the Caffe2 model in [Caffe2 91f63a236](91f63a2361fb8671e103a8d5601adec8354299b5):
```
christoph:~/caffe2-inference-test$ python run-caffe2.py MNIST.simple.caffe2 
WARNING:root:This caffe2 python run does not have GPU support. Will run in CPU only mode.
WARNING:root:Debug message: No module named caffe2_pybind11_state_gpu
Traceback (most recent call last):
  File "run-caffe2.py", line 51, in <module>
    sys.exit( main() )
  File "run-caffe2.py", line 43, in main
    predictor = caffe2.python.workspace.Predictor(init_net, predict_net)
RuntimeError: [enforce fail at operator.cc:52] blob != nullptr. op Scale: Encountered a non-existing input blob: 1 
```

Please provide guidance because I cannot tell if this is an onnx-caffe2 or a Caffe2 problem.

**Comment by bddppq** [comment]
@christoph-conrads By looking at your command line, I think you have not loaded the `init_net` (which is the file`MNIST.simple.init.caffe2` in your case).

**Comment by christoph-conrads** [comment]
I accidentally swapped `init_net` and `predict_net`. Now I have the following code based on [_Loading Pre-Trained Models_](https://caffe2.ai/docs/tutorial-loading-pre-trained-models.html):
```python
    with open(net_file, 'r') as f:
        predict_net = f.read()
    with open(init_file, 'r') as f:
        init_net = f.read()        

    predictor = caffe2.python.workspace.Predictor(init_net, predict_net)
```
When I run it, it generates the following output:
```
christoph:~/caffe2-inference-test$ python run-caffe2.py MNIST.simple.caffe2 MNIST.simple.init.caffe2 
WARNING:root:This caffe2 python run does not have GPU support. Will run in CPU only mode.
WARNING:root:Debug message: No module named caffe2_pybind11_state_gpu
Traceback (most recent call last):
  File "run-caffe2.py", line 58, in <module>
    sys.exit( main() )
  File "run-caffe2.py", line 48, in main
    predictor = caffe2.python.workspace.Predictor(init_net, predict_net)
RuntimeError: [enforce fail at operator.cc:52] blob != nullptr. op MatMul: Encountered a non-existing input blob: 1 
```

**Comment by bddppq** [comment]
You should open the file in 'rb' mode.

**Comment by christoph-conrads** [comment]
Same error as before. I tried using a pre-trained network (SqueezeNet) but then I encounter [Caffe2 issue #1294](https://github.com/caffe2/caffe2/issues/1294).

**Comment by bddppq** [comment]
Could you post the onnx pb file, the init_file and the predict_file so I can try to reproduce the error?

**Comment by christoph-conrads** [comment]
* onnx 765542db16eed8d9270c160a34ec23537f4f926c
* onnx-caffe2 9f9fac65126bb3dd7fcc39083e7e9e75fe5d4e70
* pytorch 6647475bc2ba906c452a21e3857571d4de6ecadc
* caffe2 f73c1a75f7d1095857efa55b042294b10e43f255

Minimum working example:
```python
#!/usr/bin/python
# -*- coding: utf-8 -*-

from __future__ import print_function

import caffe2.python.workspace

import numpy

import onnx
import onnx_caffe2.backend

import subprocess
import sys

import torch
import torch.autograd
import torch.nn
import torch.onnx



class SimpleModel(torch.nn.Module):
    def __init__(self):
        super(SimpleModel, self).__init__()

        self.layer = torch.nn.Linear(784, 10, bias=False)
        torch.nn.init.orthogonal(self.layer.weight)


    def forward(self, x):
        return self.layer(x)



def main():
    numpy.random.seed(1)
    torch.manual_seed(1)

    onnx_file = 'onnx-caffe2.issue48-mwe.onnx.pb'
    init_net_file = 'onnx-caffe2.issue48-mwe.init.pb'
    predict_net_file = 'onnx-caffe2.issue48-mwe.predict.pb'


    # pytorch ONNX export
    model = SimpleModel()
    model.train(False)

    x_np = numpy.ones([1, 784], dtype=numpy.float32)
    x_t = torch.from_numpy(x_np)
    x = torch.autograd.Variable(x_t)

    torch.onnx.export(model, x, onnx_file)


    # run onnx_caffe2 inference
    graph = onnx.load(onnx_file)

    onnx.checker.check_model(graph)

    model = onnx_caffe2.backend.prepare(graph)

    output = model.run(x_np)


    # onnx to caffe2
    subprocess.call( [
        'convert-onnx-to-caffe2',
        '--init-net-output', init_net_file,
        '--output', predict_net_file,
        '--', onnx_file
    ] )


    # run caffe2 inference
    with open(init_net_file, 'rb') as f:
        init_net = f.read()
    with open(predict_net_file, 'rb') as f:
        predict_net = f.read()

    predictor = caffe2.python.workspace.Predictor(init_net, predict_net)

    output = predict_net.run([x_np])



if __name__ == '__main__':
    sys.exit( main() )
```

Protobuf files generated by the minimum working example:
[onnx-caffe2.issue48-mwe.zip](https://github.com/onnx/onnx-caffe2/files/1392637/onnx-caffe2.issue48-mwe.zip)

**Comment by bddppq** [comment]
After investigation this is caused by onnx-to-caffe2 converter doesn't generate init operators for real model inputs while caffe2 predictor interface requires all input blobs are initialized. It should have been fixed by 4fb74f5.

**Comment by christoph-conrads** [comment]
Works for me with e17e80d6c10d781212bb943b2599b5b2aabef1ce.
